tensorrt_llm/_torch/auto_deploy/transformations/library/sharding.py

```diff
@@ -85,6 +92,15 @@ def _insert_sharded_matmul(
     def split_tensor(
         t: torch.Tensor, d: int = dim, r: int = rank, ws: int = world_size
     ) -> torch.Tensor:
+        # The local tensor shape has to be divisible by min_local_shape
+        max_split_size = t.shape[d] // min_local_shape
```

Comment:
The `split_tensor` function is used as a load hook outside the context of the `_insert_sharded_matmul`. So you have to "bind" `min_local_shape` from the outer context to this local function definition so it can later retrieve the correct value

```suggestion
    def split_tensor(
        t: torch.Tensor, d: int = dim, r: int = rank, ws: int = world_size, min_d_shape: int = min_local_shape
    ) -> torch.Tensor:
        # The local tensor shape has to be divisible by min_d_shape
        max_split_size = t.shape[d] // min_d_shape
```